intellij/src/saros/intellij/ui/util/SafeDialogUtils.java

```diff
@@ -192,7 +213,20 @@ public static Integer showYesNoDialog(Project project, final String message, fin
         },
         ModalityState.defaultModalityState());
 
-    return response.get();
+    Integer result = response.get();
+
+    if (result == null) {
+      throw new IllegalStateException("Synchronous yes-no dialog did not provide any results");
+    }
+
+    switch (result) {
```

Comment:
return result == Messages.YES ?

I do not get the extra error handling. Is the IntelliJ API so buggy ?

Reply:
> I do not get the extra error handling. Is the IntelliJ API so buggy ?

No, but if they ever were to decide to add other exit values that require dedicated handling I would like to know about them. This is more a catch-all for unexpected API changes.